apps/backend/src/libs/types/repository.type.ts

```diff
@@ -1,7 +1,7 @@
 type Repository<T = unknown> = {
 	create(payload: unknown): Promise<T>;
 	delete(): Promise<boolean>;
-	find(): Promise<T>;
+	find(payload: unknown): Promise<T>;
```

Comment:
`find` method should only get `id`. For searching by other field, you should create new methods for a specific repository

Reply:
I will create getByEmail()
Do I need rollback find() method ? (maybe it will be next task to implement it)

Reply:
If it is not used, roll it back

Reply:
The Repository<T = unknown> type is intended to be a common type for all future repositories. Adding a specific method like findByEmail could potentially cause issues in other repositories that don't have an email field. The same concern applies to the Service type.

Given this, should I implement new types, such as UserRepositoryType and UserServiceType, to extend the standard Repository and Service types to ensure strong typing for all methods? Or should I simply add the findByEmail method directly to the current UserRepository and UserService class?

Reply:
You should add the method directly to Repository and Service, no need to add a type for that